### ISSUE TITLE
channel.bulkDelete returns incorrect information

### ISSUE THREAD

**Issue by Mirdukkk** [discordjs/discord.js#4647]
**Please describe the problem you are having in as much detail as possible:**
I use a loop to clear more than 100 messages inside a single command call. if I specify a number greater than 100, such as 150, discord.js will return me the information that 100 messages were cleared and then 50 messages were cleared, but in fact the remaining 50 messages will not be cleared.

![oaoaommm)20-07-21-15-03-57](https://user-images.githubusercontent.com/44965055/88062137-160e4b80-cb71-11ea-8829-a1bb623085b3.png)

![oaoaommm)20-07-21-15-07-02](https://user-images.githubusercontent.com/44965055/88062180-27575800-cb71-11ea-96bc-22689bd1ed0a.png)

![oaoaommm)20-07-21-15-10-26](https://user-images.githubusercontent.com/44965055/88062211-32aa8380-cb71-11ea-9dcd-5b7917afe288.png)

**Include a reproducible code sample here, if possible:**

```js
// the code looks a little different, I simplified it so that it can be executed outside of my project
async function clear(msg, requestedToClear) {
  if (!msg) return null
  if (isNaN(requestedToClear) || requestedToClear <= 0) return msg.reply('do not joke with me')
  
  let toClearCycleNumber = Math.ceil(requestedToClear / 100),
    deletedMessagesCount = 0

  if (!msg.channel.permissionsFor(msg.guild.me).has('MANAGE_MESSAGES'))
    return msg.reply('give me permissions :c')

  for (let i = 0; i < toClearCycleNumber; i++) {
    let messagesToFetchNumber = requestedToClear - deletedMessagesCount
    if (messagesToFetchNumber > 100) messagesToFetchNumber = 100

    console.log('requested: ', messagesToFetchNumber)
    let messages = await msg.channel.messages.fetch({ before: msg.id, limit: messagesToFetchNumber })
    console.log('fetched: ', messages.size)

    await msg.channel.bulkDelete(messages, true)
      .then(deletedMessages => {
        console.log('cleared: ', deletedMessages.size)
        deletedMessagesCount += deletedMessages.size
        if(deletedMessages.size < 100) console.log('total cleared: ', deletedMessagesCount)
      })
      .catch(e => {
        i = toClearCycleNumber
        console.error(e)
      })
  }
}
```

**Further details:**

- discord.js version: 12.2.0
- Node.js version: 12.6.3
- Operating system: Windows 10 2004 & Ubuntu server 18.04 LTS
- Priority this issue should have – please be realistic and elaborate if possible: medium

**Relevant client options:**

- partials: message, reaction
- gateway intents: none
- other: none

- [ ] I have also tested the issue on latest master, commit hash:

**Comment by SmokyPlay** [comment]
Нихуя не понял, но очень интересно

**Comment by MrLivixx** [comment]
> Нихуя не понял, но очень интересно

wut?

**Comment by 0CherryTea0** [comment]
Ban SmokyPlay pls

**Comment by aue-tripenko** [comment]
кто согласен что смоки ебанутый - ставим класс

**Comment by Mirdukkk** [comment]
okay, I found the problem. MessageManager.fetch() returns a collection of messages, but some of them are returned again after deletion, as if they were not deleted. I tried not to add them to the cache, but it didn't help. I decided that I would use the ID of the message that was the last in the collection of deleted messages as the starting point, and it worked. since the problem was not on discord's side.js, I'm closing this issue.
